server/aesdsocket.c

```diff
@@ -0,0 +1,338 @@
+// Nicholas Buckley AESDsocket socket base program!
+
+// change found on stack overflow to allow netdb import to work! (It means to use POSIX 2004 standard defs)
+#define _XOPEN_SOURCE 600
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <netdb.h>
+#include <stddef.h>
+#include <syslog.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <signal.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <sys/stat.h>
+
+
+#define BUFFER_SIZE 1024
+#define DATA_PATH ("/var/tmp/aesdsocketdata")
+#define PORT_NUM ("9000")
+
+
+// addrinfo structure for reference gotten from lecture
+/*
+struct addrinfo {
+    int ai_flags;
+    int ai_family;
+    int ai_socktype;
+    int ai_protocol;
+    size_t ai_addrlen;
+    struct sockaddr *ai_addr;
+    char *ai_canonname;
+
+    struct addrinfo *ai_next;
+};*/
+
+
+// Setup global file pointers and runAsDaemon
+int sockfd;
+int openConSock;
+int writefp;
+int readfp;
+
+
+bool runAsDaemon = false;
+
+/* simple clean routine called by signal handler for SIGINT and SIGTERM to close all 
+*  file pointers and sockets, remove the file writen to and from and close the logging
+*  this function was and sig handler skeletons were gotten from bing chat gpt
+*  "Gracefully exits when SIGINT or SIGTERM is received, completing any open connection operations,
+*  closing any open sockets, and deleting the file /var/tmp/aesdsocketdata."
+*/
+void cleanUp(void)
+{
+    if (writefp != -1) 
+    {
+        close(writefp);
+        writefp = -1;
+    }
+
+    if (readfp != -1) 
+    {
+        close(readfp);
+        readfp = -1;
+    }
+
+    if (sockfd != -1) 
+    {
+        shutdown(sockfd,SHUT_RDWR);
+        close(sockfd);
+        sockfd = -1;
+    }
+
+    if (openConSock != -1) 
+    {
+        shutdown(openConSock,SHUT_RDWR);
+        close(openConSock);
+        openConSock = -1;
+    }
+
+    syslog(LOG_INFO,"Caught signal, exiting");
+
+    closelog();
+    remove(DATA_PATH);
+}
+
+void exitSigHandler(int signo) 
+{
+    if (signo == SIGINT || signo == SIGTERM) 
+    {
+        cleanUp();
+        exit(0);
```

Comment:
@Nicholas-Buckley77 is all of this safe to do from signal handler?  Consider a design where you can just request exit here and handle cleanup from the main loop instead.